### PR TITLE
bugfix:fix FileConfiguration config listener logic 

### LINKED ISSUE #771: FileConfiguration couldn't notify the listeners
- [ ] I have searched the [issues](https://github.com/alibaba/fescar/issues) of this repository and believe that this is not a duplicate.

### Ⅱ. Describe what happened
```
public void run() {
            while (true) {
                try {
                    Map<String, List<ConfigChangeListener>> configListenerMap;
                    if (null != dataId && null != listener) {
                        configListenerMap = new ConcurrentHashMap<>(8);
                        configListenerMap.put(dataId, new ArrayList<>());
                        configListenerMap.get(dataId).add(listener);
                    } else {
                        configListenerMap = configListenersMap;
                    }
                    for (Map.Entry<String, List<ConfigChangeListener>> entry : configListenerMap.entrySet()) {
                        String configId = entry.getKey();
                        String currentConfig = getConfig(configId);
                        if (null != currentConfig && currentConfig.equals(listenedConfigMap.get(configId))) {
                            listenedConfigMap.put(configId, currentConfig);
                            notifyAllListener(configId, configListenerMap.get(configId));

                        } else if (null == currentConfig && null != listenedConfigMap.get(configId)) {
                            notifyAllListener(configId, configListenerMap.get(configId));
                        }
                    }
                    Thread.sleep(LISTENER_CONFIG_INTERNAL);
                } catch (Exception exx) {
                    LOGGER.error(exx.getMessage());
                }

            }
```
if the config with some configid is not null,  but the listenedConfigMap.get(configId) is null, the code listenedConfigMap.put(configId, currentConfig) will never run, so the method notfyAllListener will never be called

### Ⅲ. Describe what you expected to happen
as the config value changed, the listener who watch these should always be notify
### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by liujianhuiouc**
### Ⅰ. Describe what this PR did
when listeners change, add or remove the config correspond to the config id, as the config change, it mean current config different from the config in listenconfigmap, the listener match that config should be called

### CODE REVIEW

File: config/src/main/java/com/alibaba/fescar/config/FileConfiguration.java [hunk 1]
```diff
@@ -123,6 +124,7 @@ public boolean removeConfig(String dataId, long timeoutMills) {
     public void addConfigListener(String dataId, ConfigChangeListener listener) {
         configListenersMap.putIfAbsent(dataId, new ArrayList<ConfigChangeListener>());
         configListenersMap.get(dataId).add(listener);
+        listenedConfigMap.putIfAbsent(dataId, getConfig(dataId));
```
Comment (CoffeeLatte007) [inline config/src/main/java/com/alibaba/fescar/config/FileConfiguration.java]: maybe put?

Comment (CoffeeLatte007) [reply config/src/main/java/com/alibaba/fescar/config/FileConfiguration.java]: if have many listeners and  this value has been updated. the data is old value

Comment (liujianhuiouc) [reply config/src/main/java/com/alibaba/fescar/config/FileConfiguration.java]: if exists listener, the value in listenedconfigmap will be update in the ConfigChangeRunnable,

Comment (CoffeeLatte007) [reply config/src/main/java/com/alibaba/fescar/config/FileConfiguration.java]: > if exists listener, the value in listenedconfigmap will be update in the ConfigChangeRunnable,

yes u are right

**Comment by liujianhuiouc** [comment]
> Please modify the title.it will be a commit log.please refer others PR.

done

**Pull request merged**